### ISSUE TITLE
Software::License installation issue

### ISSUE THREAD

**Issue by mydogonfire** [Perl-Toolchain-Gang/Software-License#86]
I tried to install the Software::License on Mac Ventura OS. Getting the following compatibility issues.
do you know if the next version will be able to support the following packages? thanks.

~~~
#     Module              Want Have
#     ------------------- ---- ----
#     ExtUtils::MakeMaker 6.78 7.34
#     JSON::PP 2.27300 4.02
#     Test::More          0.96 1.302162
#     CPAN::Meta 2.120900 2.150010
#     perl           5.006 5.030003
~~~

**Comment by karenetheridge** [comment]
It looks like you have a version greater than the necessary version, for all of those modules. Are you sure that the installation failed?

**Comment by Leont** [comment]
You may want to paste the full output so we can actually analyze what's going on.

Also, you may want to put it between `~~~~` lines so that the markup doesn't do weird things.

**Comment by mydogonfire** [comment]
I tested with another computer, once upgraded to Ventura OS (Mac). The perl version is 5.30.3 and getting same errors.

Running install for module 'Software::License'
Checksum for /Users/admin/.cpan/sources/authors/id/L/LE/LEONT/Software-License-0.104003.tar.gz ok
Configuring L/LE/LEONT/Software-License-0.104003.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Warning: prerequisite Data::Section 0 not found.
Warning: prerequisite Text::Template 0 not found.
Generating a Unix-style Makefile
Writing Makefile for Software::License
Writing MYMETA.yml and MYMETA.json
  LEONT/Software-License-0.104003.tar.gz
  /usr/bin/perl Makefile.PL -- OK
Running make for L/LE/LEONT/Software-License-0.104003.tar.gz
---- Unsatisfied dependencies detected during ----
----  LEONT/Software-License-0.104003.tar.gz  ----
    Data::Section [requires]
    Text::Template [requires]
Running install for module 'Data::Section'
Checksum for /Users/admin/.cpan/sources/authors/id/R/RJ/RJBS/Data-Section-0.200008.tar.gz ok
Configuring R/RJ/RJBS/Data-Section-0.200008.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Warning: prerequisite Test::FailWarnings 0 not found.
Generating a Unix-style Makefile
Writing Makefile for Data::Section
Writing MYMETA.yml and MYMETA.json
  RJBS/Data-Section-0.200008.tar.gz
  /usr/bin/perl Makefile.PL -- OK
Running make for R/RJ/RJBS/Data-Section-0.200008.tar.gz
---- Unsatisfied dependencies detected during ----
----     RJBS/Data-Section-0.200008.tar.gz    ----
    Test::FailWarnings [build_requires]
Running install for module 'Test::FailWarnings'
Checksum for /Users/admin/.cpan/sources/authors/id/D/DA/DAGOLDEN/Test-FailWarnings-0.008.tar.gz ok
Configuring D/DA/DAGOLDEN/Test-FailWarnings-0.008.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Test::FailWarnings
Writing MYMETA.yml and MYMETA.json
  DAGOLDEN/Test-FailWarnings-0.008.tar.gz
  /usr/bin/perl Makefile.PL -- OK
Running make for D/DA/DAGOLDEN/Test-FailWarnings-0.008.tar.gz
cp lib/Test/FailWarnings.pm blib/lib/Test/FailWarnings.pm
Manifying 1 pod document
  DAGOLDEN/Test-FailWarnings-0.008.tar.gz
  /usr/bin/make -- OK
Running make test for DAGOLDEN/Test-FailWarnings-0.008.tar.gz
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-compile.t ......... ok   
t/00-report-prereqs.t .. # Prerequisite Report:
#    Version Module               
#   -------- ---------------------
#       0.48 Capture::Tiny        
#       1.50 Carp                 
#       3.78 Cwd                  
#       7.34 ExtUtils::MakeMaker  
#       3.78 File::Spec           
#       3.78 File::Spec::Functions
#     0.2309 File::Temp           
#       1.40 IO::Handle           
#       1.20 IPC::Open3           
#       1.50 List::Util           
#   1.302162 Test::More           
#       1.33 constant             
#       0.65 lib                  
#       1.11 strict               
#       1.44 warnings             
t/00-report-prereqs.t .. ok   
t/failwarn.t ........... ok   
All tests successful.
Files=3, Tests=8,  1 wallclock secs ( 0.03 usr  0.02 sys +  0.77 cusr  0.32 csys =  1.14 CPU)
Result: PASS
  DAGOLDEN/Test-FailWarnings-0.008.tar.gz
  /usr/bin/make test -- OK
Running make install for DAGOLDEN/Test-FailWarnings-0.008.tar.gz
Manifying 1 pod document
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
ERROR: Can't create '/Library/Perl/5.30/Test'
mkdir /Library/Perl/5.30/Test: Permission denied at /System/Library/Perl/5.30/ExtUtils/Install.pm line 489.
 
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 at -e line 1.
make: *** [pure_site_install] Error 13
  DAGOLDEN/Test-FailWarnings-0.008.tar.gz
  /usr/bin/make install  -- NOT OK
  RJBS/Data-Section-0.200008.tar.gz
  Has already been unwrapped into directory /Users/admin/.cpan/build/Data-Section-0.200008-2
  RJBS/Data-Section-0.200008.tar.gz
  Has already been prepared
Running make for R/RJ/RJBS/Data-Section-0.200008.tar.gz
cp lib/Data/Section.pm blib/lib/Data/Section.pm
Manifying 1 pod document
  RJBS/Data-Section-0.200008.tar.gz
  /usr/bin/make -- OK
Running make test for RJBS/Data-Section-0.200008.tar.gz
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-report-prereqs.t .. # 
# Versions for all modules listed in MYMETA.json (including optional ones):
# 
# === Configure Requires ===
# 
#     Module              Want Have
#     ------------------- ---- ----
#     ExtUtils::MakeMaker 6.78 7.34
# 
# === Build Requires ===
# 
#     Module              Want Have
#     ------------------- ---- ----
#     ExtUtils::MakeMaker  any 7.34
# 
# === Test Requires ===
# 
#     Module              Want     Have
#     ------------------- ---- --------
#     ExtUtils::MakeMaker  any     7.34
#     File::Spec           any     3.78
#     Test::FailWarnings   any    0.008
#     Test::More          0.96 1.302162
#     base                 any     2.27
#     lib                  any     0.65
#     utf8                 any     1.22
# 
# === Test Recommends ===
# 
#     Module         Want     Have
#     ---------- -------- --------
#     CPAN::Meta 2.120900 2.150010
# 
# === Runtime Requires ===
# 
#     Module         Want  Have
#     ------------- ----- -----
#     Encode          any  3.01
#     MRO::Compat    0.09  0.13
#     Sub::Exporter 0.979 0.987
#     strict          any  1.11
#     warnings        any  1.44
# 
t/00-report-prereqs.t .. ok   
t/basic.t .............. ok    
t/encodings.t .......... ok   
All tests successful.
Files=3, Tests=40,  0 wallclock secs ( 0.02 usr  0.01 sys +  0.31 cusr  0.10 csys =  0.44 CPU)
Result: PASS
  RJBS/Data-Section-0.200008.tar.gz
  /usr/bin/make test -- OK
Running make install for RJBS/Data-Section-0.200008.tar.gz
Manifying 1 pod document
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
ERROR: Can't create '/Library/Perl/5.30/Data'
mkdir /Library/Perl/5.30/Data: Permission denied at /System/Library/Perl/5.30/ExtUtils/Install.pm line 489.
 
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 at -e line 1.
make: *** [pure_site_install] Error 13
  RJBS/Data-Section-0.200008.tar.gz
  /usr/bin/make install  -- NOT OK
Running install for module 'Text::Template'
Checksum for /Users/admin/.cpan/sources/authors/id/M/MS/MSCHOUT/Text-Template-1.61.tar.gz ok
Configuring M/MS/MSCHOUT/Text-Template-1.61.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Warning: prerequisite Test::More::UTF8 0 not found.
Generating a Unix-style Makefile
Writing Makefile for Text::Template
Writing MYMETA.yml and MYMETA.json
  MSCHOUT/Text-Template-1.61.tar.gz
  /usr/bin/perl Makefile.PL -- OK
Running make for M/MS/MSCHOUT/Text-Template-1.61.tar.gz
---- Unsatisfied dependencies detected during ----
----     MSCHOUT/Text-Template-1.61.tar.gz    ----
    Test::More::UTF8 [build_requires]
Running install for module 'Test::More::UTF8'
Checksum for /Users/admin/.cpan/sources/authors/id/M/MO/MONS/Test-More-UTF8-0.05.tar.gz ok
Configuring M/MO/MONS/Test-More-UTF8-0.05.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Test::More::UTF8
Writing MYMETA.yml and MYMETA.json
  MONS/Test-More-UTF8-0.05.tar.gz
  /usr/bin/perl Makefile.PL -- OK
Running make for M/MO/MONS/Test-More-UTF8-0.05.tar.gz
cp lib/Test/More/UTF8.pm blib/lib/Test/More/UTF8.pm
Manifying 1 pod document
  MONS/Test-More-UTF8-0.05.tar.gz
  /usr/bin/make -- OK
Running make test for MONS/Test-More-UTF8-0.05.tar.gz
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-load.t ....... 1/1 # Testing Test::More::UTF8 0.05, Perl 5.030003, /usr/bin/perl
t/00-load.t ....... ok   
t/01-usage.t ...... 1/4 # а
t/01-usage.t ...... ok   
t/02-usage.t ...... 1/4 # а
t/02-usage.t ...... ok   
t/03-usage.t ...... 1/4 # а
t/03-usage.t ...... ok   
t/pod-coverage.t .. skipped: Test::Pod::Coverage 1.08 required for testing POD coverage
t/pod.t ........... ok   
All tests successful.
Files=6, Tests=14,  1 wallclock secs ( 0.03 usr  0.01 sys +  0.37 cusr  0.15 csys =  0.56 CPU)
Result: PASS
  MONS/Test-More-UTF8-0.05.tar.gz
  /usr/bin/make test -- OK
Running make install for MONS/Test-More-UTF8-0.05.tar.gz
Manifying 1 pod document
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
ERROR: Can't create '/Library/Perl/5.30/Test/More'
mkdir /Library/Perl/5.30/Test: Permission denied at /System/Library/Perl/5.30/ExtUtils/Install.pm line 489.
 
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 at -e line 1.
make: *** [pure_site_install] Error 13
  MONS/Test-More-UTF8-0.05.tar.gz
  /usr/bin/make install  -- NOT OK
  MSCHOUT/Text-Template-1.61.tar.gz
  Has already been unwrapped into directory /Users/admin/.cpan/build/Text-Template-1.61-2
  MSCHOUT/Text-Template-1.61.tar.gz
  Has already been prepared
Running make for M/MS/MSCHOUT/Text-Template-1.61.tar.gz
cp lib/Text/Template/Preprocess.pm blib/lib/Text/Template/Preprocess.pm
cp lib/Text/Template.pm blib/lib/Text/Template.pm
Manifying 2 pod documents
  MSCHOUT/Text-Template-1.61.tar.gz
  /usr/bin/make -- OK
Running make test for MSCHOUT/Text-Template-1.61.tar.gz
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/author-pod-syntax.t .. skipped: these tests are for testing by the author
t/author-signature.t ... skipped: these tests are for testing by the author
t/basic.t .............. ok     
t/broken.t ............. ok   
t/delimiters.t ......... ok     
t/error.t .............. ok   
t/exported.t ........... ok   
t/hash.t ............... ok     
t/inline-comment.t ..... ok   
t/nested-tags.t ........ ok   
t/ofh.t ................ ok   
t/out.t ................ ok   
t/prepend.t ............ ok     
t/preprocess.t ......... ok   
t/rt29928.t ............ ok   
t/safe.t ............... ok     
t/safe2.t .............. ok     
t/safe3.t .............. ok   
t/strict.t ............. ok   
t/taint.t .............. ok     
t/template-encoding.t .. ok   
t/warnings.t ........... ok   
All tests successful.
Files=22, Tests=184,  2 wallclock secs ( 0.07 usr  0.04 sys +  1.34 cusr  0.53 csys =  1.98 CPU)
Result: PASS
  MSCHOUT/Text-Template-1.61.tar.gz
  /usr/bin/make test -- OK
Running make install for MSCHOUT/Text-Template-1.61.tar.gz
Manifying 2 pod documents
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
ERROR: Can't create '/Library/Perl/5.30/Text'
mkdir /Library/Perl/5.30/Text: Permission denied at /System/Library/Perl/5.30/ExtUtils/Install.pm line 489.
 
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 at -e line 1.
make: *** [pure_site_install] Error 13
  MSCHOUT/Text-Template-1.61.tar.gz
  /usr/bin/make install  -- NOT OK
  LEONT/Software-License-0.104003.tar.gz
  Has already been unwrapped into directory /Users/admin/.cpan/build/Software-License-0.104003-2
  LEONT/Software-License-0.104003.tar.gz
  Has already been prepared
Running make for L/LE/LEONT/Software-License-0.104003.tar.gz
cp lib/Software/License/Perl_5.pm blib/lib/Software/License/Perl_5.pm
cp lib/Software/License/Mozilla_2_0.pm blib/lib/Software/License/Mozilla_2_0.pm
cp lib/Software/License/None.pm blib/lib/Software/License/None.pm
cp lib/Software/License/SSLeay.pm blib/lib/Software/License/SSLeay.pm
cp lib/Software/License/EUPL_1_1.pm blib/lib/Software/License/EUPL_1_1.pm
cp lib/Software/License/Apache_1_1.pm blib/lib/Software/License/Apache_1_1.pm
cp lib/Software/License/GPL_1.pm blib/lib/Software/License/GPL_1.pm
cp lib/Software/License/AGPL_3.pm blib/lib/Software/License/AGPL_3.pm
cp lib/Software/License/Artistic_1_0_Perl.pm blib/lib/Software/License/Artistic_1_0_Perl.pm
cp lib/Software/License/CC0_1_0.pm blib/lib/Software/License/CC0_1_0.pm
cp lib/Software/License/Artistic_1_0.pm blib/lib/Software/License/Artistic_1_0.pm
cp lib/Software/License/Mozilla_1_1.pm blib/lib/Software/License/Mozilla_1_1.pm
cp lib/Software/License/Sun.pm blib/lib/Software/License/Sun.pm
cp lib/Software/License/OpenSSL.pm blib/lib/Software/License/OpenSSL.pm
cp lib/Software/License/MIT.pm blib/lib/Software/License/MIT.pm
cp lib/Software/License/FreeBSD.pm blib/lib/Software/License/FreeBSD.pm
cp lib/Software/License/GFDL_1_3.pm blib/lib/Software/License/GFDL_1_3.pm
cp lib/Software/License.pm blib/lib/Software/License.pm
cp lib/Software/License/PostgreSQL.pm blib/lib/Software/License/PostgreSQL.pm
cp lib/Software/License/Artistic_2_0.pm blib/lib/Software/License/Artistic_2_0.pm
cp lib/Software/License/GFDL_1_2.pm blib/lib/Software/License/GFDL_1_2.pm
cp lib/Software/License/QPL_1_0.pm blib/lib/Software/License/QPL_1_0.pm
cp lib/Software/LicenseUtils.pm blib/lib/Software/LicenseUtils.pm
cp lib/Software/License/Mozilla_1_0.pm blib/lib/Software/License/Mozilla_1_0.pm
cp lib/Software/License/LGPL_2_1.pm blib/lib/Software/License/LGPL_2_1.pm
cp lib/Software/License/Custom.pm blib/lib/Software/License/Custom.pm
cp lib/Software/License/Apache_2_0.pm blib/lib/Software/License/Apache_2_0.pm
cp lib/Software/License/EUPL_1_2.pm blib/lib/Software/License/EUPL_1_2.pm
cp lib/Software/License/LGPL_3_0.pm blib/lib/Software/License/LGPL_3_0.pm
cp lib/Software/License/Zlib.pm blib/lib/Software/License/Zlib.pm
cp lib/Software/License/ISC.pm blib/lib/Software/License/ISC.pm
cp lib/Software/License/BSD.pm blib/lib/Software/License/BSD.pm
cp lib/Software/License/GPL_2.pm blib/lib/Software/License/GPL_2.pm
cp lib/Software/License/GPL_3.pm blib/lib/Software/License/GPL_3.pm
Manifying 34 pod documents
  LEONT/Software-License-0.104003.tar.gz
  /usr/bin/make -- OK
Running make test for LEONT/Software-License-0.104003.tar.gz
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-report-prereqs.t ....... # 
# Versions for all modules listed in MYMETA.json (including optional ones):
# 
# === Configure Requires ===
# 
#     Module              Want Have
#     ------------------- ---- ----
#     ExtUtils::MakeMaker 6.78 7.34
# 
# === Configure Suggests ===
# 
#     Module      Want Have
#     -------- ------- ----
#     JSON::PP 2.27300 4.02
# 
# === Build Requires ===
# 
#     Module              Want Have
#     ------------------- ---- ----
#     ExtUtils::MakeMaker  any 7.34
# 
# === Test Requires ===
# 
#     Module              Want     Have
#     ------------------- ---- --------
#     ExtUtils::MakeMaker  any     7.34
#     File::Spec           any     3.78
#     Test::More          0.96 1.302162
#     Try::Tiny            any     0.30
# 
# === Test Recommends ===
# 
#     Module         Want     Have
#     ---------- -------- --------
#     CPAN::Meta 2.120900 2.150010
# 
# === Runtime Requires ===
# 
#     Module          Want     Have
#     -------------- ----- --------
#     Carp             any     1.50
#     Data::Section    any 0.200008
#     File::Spec       any     3.78
#     IO::Dir          any     1.40
#     Module::Load     any     0.34
#     Text::Template   any     1.61
#     parent           any    0.237
#     perl           5.006 5.030003
#     strict           any     1.11
#     utf8             any     1.22
#     warnings         any     1.44
# 
t/00-report-prereqs.t ....... ok   
t/apache_2_0.t .............. ok   
t/basic.t ................... ok   
t/bsd-url.t ................. ok   
t/creative_commons.t ........ skipped: requires Software::License::CCpack to test this
t/custom.t .................. ok   
t/from-spdx-expression.t .... ok   
t/guess_license_from_pod.t .. ok   
t/guess_meta_license.t ...... ok     
t/meta-names.t .............. ok    
t/program.t ................. ok   
t/round-trip.t .............. 1/22 
#   Failed test 'Artistic_1_0: ...and we got it right'
#   at t/round-trip.t line 31.
#          got: 'Software::License::Artistic_1_0_Perl'
#     expected: 'Software::License::Artistic_1_0'
# Looks like you failed 1 test of 22.
t/round-trip.t .............. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/22 subtests 
t/short_name.t .............. ok   
t/spdx-expression.t ......... ok     
t/two-dots.t ................ ok    
t/utils.t ................... ok   
 
Test Summary Report
-------------------
t/round-trip.t            (Wstat: 256 Tests: 22 Failed: 1)
  Failed test:  4
  Non-zero exit status: 1
Files=16, Tests=232,  3 wallclock secs ( 0.06 usr  0.03 sys +  1.51 cusr  0.54 csys =  2.14 CPU)
Result: FAIL
Failed 1/16 test programs. 1/232 subtests failed.
make: *** [test_dynamic] Error 255
  LEONT/Software-License-0.104003.tar.gz
  /usr/bin/make test -- NOT OK
//hint// to see the cpan-testers results for installing this module, try:
  reports LEONT/Software-License-0.104003.tar.gz

**Comment by Leont** [comment]
(I took the liberty of properly formatting your posts for readability, you need to quote blocks like that in between `~~~` lines)

**Comment by Leont** [comment]
You have your answer at
```
ERROR: Can't create '/Library/Perl/5.30/Test'
mkdir /Library/Perl/5.30/Test: Permission denied at /System/Library/Perl/5.30/ExtUtils/Install.pm line 489.
```
You can't install to a system directory as your user. You either need to set up a `local::lib` to install to, build your own perl (e.g. with `perlbrew`) or need to run `cpan` with `sudo` (not generally recommended). Homebrew may also be able to help but I don't know much about it. In any case this is not a `Software::License` issue.

**Comment by Leont** [comment]
```
t/round-trip.t .............. 1/22 
#   Failed test 'Artistic_1_0: ...and we got it right'
#   at t/round-trip.t line 31.
#          got: 'Software::License::Artistic_1_0_Perl'
#     expected: 'Software::License::Artistic_1_0'
# Looks like you failed 1 test of 22.
t/round-trip.t .............. Dubious, test returned 1 (wstat 256, 0x100)
```
Actually, that is a real issue, that has been fixed by the release of Software::License 0.104004 yesterday.
